### PR TITLE
Feature/code structure

### PR DESCRIPTION
**Pull request by jaimala**
code structure branch created applying all previous review comments

### CODE REVIEW

File: app/src/main/java/com/example/testapp/ui/login/LoginActivity.kt [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.util.Log
+
+
+import androidx.lifecycle.Observer
+import com.example.testapp.R
+
+class LoginActivity : AppCompatActivity() {
```
Comment (asdheeraj) [inline app/src/main/java/com/example/testapp/ui/login/LoginActivity.kt]: Please make this a fragment. It is not good to have multiple activities for multiple functionalities. Host these fragments in single Activity (Please name this Activity to be something meaningful)

File: app/src/main/java/com/example/testapp/ui/movie/MovieActivity.kt [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package com.example.testapp.ui.movie
+
+import android.os.Bundle
+import android.view.View
+import androidx.appcompat.app.AppCompatActivity
+import com.example.testapp.R
+
+
+class MovieActivity : AppCompatActivity() {
```
Comment (asdheeraj) [inline app/src/main/java/com/example/testapp/ui/movie/MovieActivity.kt]: Please make this a fragment. It is not good to have multiple activities for multiple functionalities. Host these fragments in single Activity (Please name this Activity to be something meaningful)

**Review by asdheeraj — COMMENTED** [review]
Please address the comments. Also rebase ur branch with development and please squash ur commits. There should only be one commit above development.

File: app/src/main/java/com/example/testapp/ui/movie/MovieActivity.kt (hunk 2 above)
Comment (jaimala) [reply app/src/main/java/com/example/testapp/ui/movie/MovieActivity.kt]: changed to fragment

File: app/src/main/java/com/example/testapp/ui/login/LoginActivity.kt (hunk 1 above)
Comment (jaimala) [reply app/src/main/java/com/example/testapp/ui/login/LoginActivity.kt]: changed to fragment

File: app/src/main/java/com/example/testapp/ui/MainActivity.kt [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package com.example.testapp.ui
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import com.example.testapp.R
+
+class MainActivity : AppCompatActivity() {
```
Comment (asdheeraj) [inline app/src/main/java/com/example/testapp/ui/MainActivity.kt]: Rename this MainActivity to something meaningful as mentioned in the previous comments. Likse launchActivity, DashboardActivity etc

File: app/src/main/java/com/example/testapp/util/Util.kt [hunk 4]
```diff
@@ -0,0 +1,6 @@
+package com.example.testapp.util
+
+
+class Util {
```
Comment (asdheeraj) [inline app/src/main/java/com/example/testapp/util/Util.kt]: this should a object(static class) in kotlin for easy access from any class

**Review by asdheeraj — COMMENTED** [review]
Please rebase ur branch. I see multiple commits. Please squash ur commits. I have mentioned this in my previous comment already

File: app/src/main/java/com/example/testapp/ui/MainActivity.kt (hunk 3 above)
Comment (jaimala) [reply app/src/main/java/com/example/testapp/ui/MainActivity.kt]: renamed

File: app/src/main/java/com/example/testapp/util/Util.kt (hunk 4 above)
Comment (jaimala) [reply app/src/main/java/com/example/testapp/util/Util.kt]: companion object

**Review by asdheeraj — COMMENTED** [review]
Ur commits are not Squashed. Please Squash. Im mentioning this from the past three comments. Please rebase ur branch with dev and squash ur commits. there shoudl be onlhy one commit above development.
Please check ur mail for refernce link
in that link, instead of master, rebase to development

**Comment by jaimala** [comment]
rebased with develop and squashed commits

**Review by asdheeraj — DISMISSED** [review]

File: app/src/main/java/com/example/testapp/util/Util.kt (hunk 4 above)
Comment (asdheeraj) [reply app/src/main/java/com/example/testapp/util/Util.kt]: make the class as object not adding a companion object. this class should be a static class which is object in kotlin: THis should be singleton: Please refer this: https://discuss.kotlinlang.org/t/how-can-i-create-a-static-class-in-kotlin-static-class-not-method-or-variable/7532

**Review by asdheeraj — COMMENTED** [review]
Please address this comment

File: app/src/main/java/com/example/testapp/ui/movie/MovieActivity.kt (hunk 2 above)
Comment (asdheeraj) [reply app/src/main/java/com/example/testapp/ui/movie/MovieActivity.kt]: Still this is an activity instead of fragment

**Review by asdheeraj — COMMENTED** [review]
Address this comment

**Review by asdheeraj — COMMENTED** [review]
Please review all the comments and only then push ur changes. Some comments are not addressed

**Comment by jaimala** [comment]
implemented changes for MovieActivity, Util class, removing functionality code and rebased code and squashed commits

**Review by asdheeraj — APPROVED** [review]

**Review by debasish1987 — APPROVED** [review]

**Pull request merged**